python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -711,23 +710,65 @@ def _compute_sys_path(self):
         :returns: Path to the current core.
         """
         # While core swapping, the Python path is not updated with the new core's Python path,
-        # so make sure the current core is at the front of the Python path for out subprocesses.
+        # so make sure the current core is at the front of the Python path for our subprocesses.
         python_folder = sgtk.bootstrap.ToolkitManager.get_core_python_path()
-        logger.debug("Adding %s to sys.path for subprocesses.", python_folder)
+        logger.info("Adding %s to sys.path for subprocesses.", python_folder)
         return [python_folder] + sys.path
 
     ###########################################################################
     # Internal methods
 
     @sgtk.LogManager.log_timing
-    def _cache_actions(self, data, config_data):
+    def _cache_actions(self, data, config_data, cached_contents_hash=None, async=False):
         """
         Triggers the caching or recaching of engine commands.
 
         :param dict data: The data passed down from the wss client.
         :param dict config_data: A dictionary that contains, at a minimum,
             "lookup_hash", "contents_hash", "descriptor", and "entity" keys.
+        :param str cached_contents_hash: If given, represents the currently
+            cached contents hash for the configuration. If this is the same as
+            a newly-generated contents hash built prior to caching, then the
+            caching process will stop and exit without doing any additional
+            work. This represents the situation where we've been asked to
+            re-cache actions, but we then prove that the existing cached data
+            is still valid.
+        :param bool async: Indicates whether the caching routine should be
+            run asynchronously.
         """
+        descriptor = config_data["descriptor"]
+        lookup_hash = config_data["lookup_hash"]
+
+        if async:
```

Comment:
This is where the thread is spawned that will check the validity of the cached data and then potentially recaching.